src/proc.cpp

```diff
@@ -453,40 +453,34 @@ static void process_mark_finished_children(parser_t &parser, bool block_ok) {
     reap_disowned_pids();
 }
 
-/// Given a command like "cat file", truncate it to a reasonable length.
-static wcstring truncate_command(const wcstring &cmd) {
-    const size_t max_len = 32;
-    if (cmd.size() <= max_len) {
-        // No truncation necessary.
-        return cmd;
+/// Call the fish_job_summary function with the given args.
+static void print_job_summary(parser_t &parser, wcstring_list_t args) {
+    wcstring buffer = wcstring(L"fish_job_summary");
+    for (const wcstring &arg : args) {
+        buffer.push_back(L' ');
+        buffer.append(escape_string(arg, ESCAPE_ALL));
```

Comment:
I think escaping is not desirable here - the function should receive the actual command that was run. It can be escaped with `string escape` if the function chooses to (maybe it should).

Reply:
This is building a single string for execution, so the individual arguments need to be escaped once in order to be split up correctly.

With escaping the execution string will be `fish_job_summary 3 sleep\ 10\ \&', which will correctly provide the args "3" and "sleep 10 &". Without escaping the execution string will be `fish_job_summary 3 sleep 10 &`, which in addition to providing the wrong args ""3", "sleep", "10", will also try to background the function call.

You can see from the expect based tests that the special characters are not escaped when the function prints them, so I think that covers what you want.